### PR TITLE
Hide date on single pages if not set

### LINKED ISSUE #31: Hide date on single pages when not set
Imho a page should not require a date assigned because it is (never?) changing content, it currently defaults to `0001-01-01` which is ugly.
A check would be nice to hide the date if its not set or even never show it for single pages, only posts.

### PR DESCRIPTION
**Pull request by MyIgel**
closes #31 (Hide date on single pages when not set)

### CODE REVIEW

File: layouts/_default/single.html [hunk 1]
```diff
@@ -2,9 +2,11 @@
   <div class="post">
     <h2 class="post-title"><a href="{{ .Permalink }}">{{ .Title | markdownify }}</a></h2>
     <div class="post-meta">
-      <span class="post-date">
-        {{ .Date.Format "2006-01-02" }}
-      </span>
+      {{ with .Date | default "" }}
```
Comment (panr) [inline layouts/_default/single.html]: Is it necessary to set `default` here? 🤔

Comment (MyIgel) [reply layouts/_default/single.html]: Sadly it is (I am also not that happy with it), the problem is that an empty date defaults to `0001-01-01` and an `if` or `with` statement seems to convert it to this string when checked. If you have a better solution: I'm happy to see it :)

Comment (panr) [reply layouts/_default/single.html]: Huh, I thought that `if` statement doesn't require any context or default value. I'll check it later ;-)

Comment (MyIgel) [reply layouts/_default/single.html]: I changed it to use `nil` (seems more meaningful) and nope, an `if` has the same problem

**Pull request merged**